internal/apiutils/requests.go

```diff
@@ -0,0 +1,37 @@
+package apiutils
+
+import "strings"
+
+const (
+	// adapterSeparator is the separator used to split model and adapter names
+	// in API requests.
+	//
+	// Alternatives considered:
+	//
+	// "-" (hyphen): This is a common separator in Kubernetes resource names.
+	// "." (dot): This is a common separator in model versions "llama-3.2".
+	// "/" (slash): This would be incompatible with specifying model names inbetween slashes in URL paths (i.e. "/some-api/models/<model-id>/details").
+	// ":" (colon): This might cause problems when specifying model names before colons in URL paths (see example below).
+	//
+	// See example of a path used in the Gemini API (https://ai.google.dev/gemini-api/docs/text-generation?lang=rest):
+	// "https://generativelanguage.googleapis.com/v1beta/models/gemini-1.5-flash:generateContent?key=$GOOGLE_API_KEY"
+	adapterSeparator = "_"
```

Comment:
Appreciate you sharing the alternatives as comments! I could see a future where a user needs it changed, but we can tackle it whenever that happens.